aviatrix/data_source_aviatrix_firewall_instance_images.go

```diff
@@ -73,3 +84,115 @@ func dataSourceAviatrixFirewallInstanceImagesRead(d *schema.ResourceData, meta i
 	d.SetId(vpcId)
 	return nil
 }
+
+type byVersion []string
+
+func (p byVersion) Len() int {
+	return len(p)
+}
+func (p byVersion) Swap(i, j int) {
+	p[i], p[j] = p[j], p[i]
+}
+func (p byVersion) Less(i, j int) bool {
+	if checkFirstCharacter(p[i]) == "R" {
+		if strings.Contains(p[i], "_") {
+			return versionFormat1(p[i], p[j], "_")
+		} else if strings.Contains(p[i], "-") {
+			return versionFormat1(p[i], p[j], "-")
+		} else {
+			log.Printf("need to add a new method sort this version format")
+		}
+	} else if checkFirstCharacter(p[i]) == "P" {
+		return versionFormat2(p[i], p[j], "-")
+	} else {
+		return compareVersion(p[i], p[j])
+	}
+	return false
+}
+
+//Rxx.xx-xxx.xxx
+//Rxx.xx_revx.x
+func versionFormat1(version1, version2, flag string) bool {
+	//[Rxx.xx, xxx.xxx]
```

Comment:
Not sure what this comment means. Can you be more explicit about what this is trying to say?

Reply:
remove this comment and add new comment to descripe the function